battles/mixins.py

```diff
@@ -9,12 +9,14 @@ class TrainerIsNotOpponentMixin(UserPassesTestMixin):
     redirect_field_name = None
 
     def test_func(self):
-        battle_id = self.request.GET.get('id')
-        trainer_creator = Battle.objects.get(id=battle_id).trainer_creator
         user = self.request.user
-        if trainer_creator == user:
+        battle_id = self.request.GET.get('id')
+        battle = Battle.objects.get(id=battle_id)
+        status = battle.status
+        trainer_opponent = battle.trainer_opponent
+        if ((trainer_opponent != user) or (status == 'SETTLED')):
```

Comment:
In Python you don't need parenthesis for the `if` statements, unless you're trying to isolate groups of conditions (which is not necessary) here :)

Also, sometimes it's good for clarity to extract values into shorter, more concise variables (e.g `status = battle.status`), but in the case of these 2 above, I think you could just access them directly and save 2 lines of code. It's not wrong the way you did, it's more a matter of code style, so feel free to do as you please :)

Reply:
By the way, do you have the pre-commit and linters installed in your machine/IDE? I believe this is the kind of error that these tools tend to pick up quickly ;)

Reply:
Yes, I do have the pre-commit and linters installed, the CircleCI wouldn't pass if I hadn't. Thank you for the second review, tomorrow I am going to read everything to make the improvements.

Reply:
Actually, unless there's some different configuration in your build settings that I'm not aware of, CircleCI doesn't check for linting errors, it only runs tests :)

Reply:
I said that because I remember that, when I was setting up the project at the begging, I got a CircleCI fail build because it run `prospector` and there were lint errors in the message. But sorry if I said something erroneous, I may have misunderstood.

Reply:
No worries, you actually have got a point there :) it might just be the case that whatever linter is running in the build moment is not catching these. It might even have been the case that the last time you've changed this file you commited with the `-n` at the end to skip the pre-commit process thus not catching it. But it's no big deal, I just wanted to ask because it helps a lot and saves a ton of time while developing :)

Reply:
I don't know too why it didn't catch, because I didn't use the n flag. I've never used linter before this project, so now I am seeing how helpful it can be. Thank you for explaining it.

Reply:
Done!